### ISSUE TITLE
Support for angular v10 please

### ISSUE THREAD

**Issue by bunnyvishal6** [h2qutc/angular-material-components#65]

**Comment by kosmoFDC** [comment]
please, update to angular cli 10 ! 👍

**Comment by bunnyvishal6** [comment]
> please, update to angular cli 10 ! 👍

Getting errors. Did you update to v10?

**Comment by tosehee** [comment]
bump

**Comment by bendav91** [comment]
Looks like angular 10 calendar component now emits an { value, event } object instead of just emiting a primitive value.

**Comment by bendav91** [comment]
Further to this, i've upgraded the project to angular 10, fixed this issue and submitted a PR.

tested and verified working.

**Comment by tosehee** [comment]
how can I pick this up now?

**Comment by daveaberl** [comment]
I think there is another problem with angular material 10 that prevents the selected date to be highlighted.
In Material 10 the input-property selectedValue (of the component mat-calendar-body) was changed to startValue and endValue (to support DateRangePicker).

**Comment by bendav91** [comment]
> I think there is another problem with angular material 10 that prevents the selected date to be highlighted.
> In Material 10 the input-property selectedValue (of the component mat-calendar-body) was changed to startValue and endValue (to support DateRangePicker).

Yes i had noticed! Sneaky buggers. They must be introducing range based date selection soon, which would take the SaturnDatePicker out the equation.

I managed to upgrade the project (really hacky and quick) to ng10 and got this working. However, the earlier PR i submitted with this made Travis CI cry. However, i can confirm if you were to take my fork on master, and go through the upgrade process to ng 10, and add those new bindings - it works! but the project is 3 versions behind...

**Comment by tosehee** [comment]
When can we expect the release with this fix?

**Comment by h2qutc** [comment]
Hi everyone, thanks for using the lib. 
@tosehee  A new version with support angular v10 will be released within 3 days.

**Comment by tosehee** [comment]
Thanks

**Comment by sambhavmalhotra** [comment]
Is there some workaround I can do for the time being to keep my app working till we have the new release?

**Comment by gultyaev** [comment]
> Is there some workaround I can do for the time being to keep my app working till we have the new release?

You can always use older Angular versions.

**Comment by sambhavmalhotra** [comment]
Already upgraded the whole application, would not want to go back to older for a few days.

**Comment by testolame** [comment]
> Already upgraded the whole application, would not want to go back to the older one for a few days.

You can downgrade only angular material, worked for me.

`npm install @angular/material@9.2.4`

**Comment by sambhavmalhotra** [comment]
> > Already upgraded the whole application, would not want to go back to the older one for a few days.
> 
> You can downgrade only angular material, worked for me.
> 
> `npm install @angular/material@9.2.4`

Thanks. Works well till we get the new release.

**Comment by wilson3q** [comment]
use this patch can make it work:

@@ -269,7 +269,7 @@
             /** @type {?} */
             const selectedMonth = this._dateAdapter.getMonth(this.activeDate);
             /** @type {?} */
-            const selectedDate = this._dateAdapter.createDate(selectedYear, selectedMonth, date);
+            const selectedDate = this._dateAdapter.createDate(selectedYear, selectedMonth, date.value);
             this.selectedChange.emit(selectedDate);
         }
         this._userSelection.emit();

**Comment by wilson3q** [comment]
but the selected date is not highlighted properly. that is the only issue I have for angular 10.

**Comment by wilson3q** [comment]
source code patch:

-- a/projects/datetime-picker/src/lib/month-view.ts
+++ b/projects/datetime-picker/src/lib/month-view.ts
@@ -133,7 +133,7 @@ export class NgxMatMonthView<D> implements AfterContentInit {
     if (this._selectedDate != date) {
       const selectedYear = this._dateAdapter.getYear(this.activeDate);
       const selectedMonth = this._dateAdapter.getMonth(this.activeDate);
-      const selectedDate = this._dateAdapter.createDate(selectedYear, selectedMonth, date);
+      const selectedDate = this._dateAdapter.createDate(selectedYear, selectedMonth, date.value);
 
       this.selectedChange.emit(selectedDate);
     }

**Comment by pepe19000** [comment]
I'm patienlty waiting for the new version. 
I love this lib.

**Comment by tosehee** [comment]
Any update on this? I thought a new version will be out in 3 days, but it's now 6 days passed.

**Comment by h2qutc** [comment]
Hi, a new version **@angular-material-components/datetime-picker@4.0.0-beta.1** has been released with support Angular v10.
Check it out by using
`npm install --save @angular-material-components/datetime-picker@4.0.0-beta.1`

**Comment by nolafs** [comment]
Hi,

Thanks but getting import errors in my shared module now.

  ERROR in node_modules/@angular-material-components/datetime-picker/lib/datetime-picker.module.d.ts:1:22 - error NG6002: Appears in the NgModule.imports of SharedModule, but could not be resolved to an NgModule class.

Any ideas?

Cheers
Olaf

**Comment by llyrowenjones** [comment]
After updating to 4.0.0 I'm getting the following error;

ERROR in Unexpected value 'NgxMatTimepickerModule in ~/node_modules/@angular-material-components/datetime-picker/lib/timepicker.module.d.ts' imported by the module '~Module in ~.module.ts'. Please add a @NgModule annotation.

Had to revert back to 2.0.4 and Material 9 to get this working until there is a fix for this.

**Comment by gultyaev** [comment]
@h2qutc Is it possible to fix peer dependencies?
```
"peerDependencies": {
    "@angular/platform-browser": "0.0.0-NG",
    "@angular/common": "0.0.0-NG",
    "@angular/core": "0.0.0-NG",
    "@angular/forms": "0.0.0-NG",
    "@angular/material": "0.0.0-PLACEHOLDER",
    "@angular/cdk": "0.0.0-PLACEHOLDER",
    "tslib": "0.0.0-TSLIB"
  }```

It shows lots of warnings after package install and forces `ng update` to throw error due to dependency incompatibility (although it should complain about @angular/material)

**Comment by h2qutc** [comment]
@llyrowenjones  a new version @angular-material-components/datetime-picker@4.0.0 has been released to fix the above bugs 
```
npm install --save @angular-material-components/datetime-picker@4.0.0
```

@gultyaev  The peer dependencies is fixed in the version 4.0.0.

**Comment by gultyaev** [comment]
@h2qutc after installing "@angular-material-components/color-picker": "^4.0.0", I had warnings
```
npm WARN ngx-color-picker@9.1.0 requires a peer of @angular/common@^9.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-color-picker@9.1.0 requires a peer of @angular/core@^9.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-color-picker@9.1.0 requires a peer of @angular/forms@^9.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-color-picker@9.1.0 requires a peer of tslib@^1.10.0 but none is installed. You must install peer dependencies yourself.
```

**Comment by jpike88** [comment]
```
ERROR in node_modules/@angular-material-components/datetime-picker/lib/datetime-picker.module.d.ts:1:22 - error NG6002: Appears in the NgModule.imports of SharedModule, but could not be resolved to an NgModule class.
```

This error is due to the ngcc being unable to compile this library correctly, it can be bypassed by disabling ivy. 

@h2qutc will need to ensure this library compiles properly with Angular 10 + ivy compiler enabled, in order to fix that issue.

**Comment by llyrowenjones** [comment]
Ivy is disabled in my current project, running Angular 10 and Material 10, and now works as expected.

**Comment by h2qutc** [comment]
@jpike88 Yes, the temporary solution is disabling Ivy compiler. I will try to release an another lib compiled property with Angular 10 + Ivy compiler enabled.

**Comment by kosmoFDC** [comment]
Thank you so much for your effort and work !!.
I would like to know when do you think to have another compilation with ivy compiler enabled aprox? :-) :-)

Thank you once again !!

Frank

**Comment by rscherer** [comment]
Hi, I'm getting the following errors (Angular 10.0.4 + Material 10.0.2, IvyEnabled, library v4.0.0)
`ERROR TypeError: Class constructor DateAdapter cannot be invoked without 'new'
    at DatetimeAdapter.NgxMatDateAdapter [as constructor] (angular-material-components-datetime-picker.js:342)
    at DatetimeAdapter.NgxMatNativeDateAdapter [as constructor] (angular-material-components-datetime-picker.js:5964)
    at new DatetimeAdapter (datetime-adapter.ts:9)`

With the following implementation:
`
export class MatDateAdapter extends NgxMatNativeDateAdapter {
public parse(...)
public format(...)
}
`

And on component:
` providers: [
        {
            provide: NgxMatDateAdapter,
            useClass: DatetimeAdapter,
            deps: [MAT_DATE_LOCALE, MAT_MOMENT_DATE_ADAPTER_OPTIONS]
        },
        { provide: NGX_MAT_DATE_FORMATS, useValue: CUSTOM_DATE_FORMATS }
    ]`

It was working perfectly fine on v9
Is it related to v10, or am I doing something wrong?

**Comment by jpike88** [comment]
@rscherer I'm getting this issue too. Looks like @h2qutc has other priorities so I'd recommend forking and taking care of this yourself... that's what I've been doing

**Comment by jpike88** [comment]
There's an open PR for this in #71, no comments, no actions. I solved the issue a different way (hacky but works), but the PR's code is much better.

**Comment by h2qutc** [comment]
@jpike88 @rscherer  I'm trying to resolve the problem and release a new version within 2 days.

**Comment by jpike88** [comment]
Are you able to reproduce the error ```ERROR TypeError: Class constructor DateAdapter cannot be invoked without 'new' at DatetimeAdapter.NgxMatDateAdapter```?

**Comment by jpike88** [comment]
I'm creating another issue specifically about this error so it's clear what needs to be sorted for this to work on Angular 10.

**Comment by kosmoFDC** [comment]
Please h2qutc ,
Could you add Ivy compiler enable in this new version that you will do ?  :-) :-) 👍 

Thank you 

Frank

**Comment by jpike88** [comment]
Ivy compiler shouldn't be a priority for next version, it needs to work without Ivy first, and it's breaking on V10. Ivy is a pain and needs to be treated as its own thing, for a future version

**Comment by jpike88** [comment]
@rscherer I think the problem is this:

https://github.com/angular/angular-cli/issues/18067

You'll need to force usage of the ES2015 bundle format.

**Comment by h2qutc** [comment]
@jpike88 @rscherer @kosmoFDC  The new version [@angular-material-components/datepicker v4.0.1](https://www.npmjs.com/package/@angular-material-components/datetime-picker) has already been released that works with Ivy compiler. Can you check it out and let me know whether it works fine plz?

**Comment by jpike88** [comment]
Sorry but I've switched to [https://github.com/danielmoncada/date-time-picker](https://github.com/danielmoncada/date-time-picker), it's better maintained and higher quality

**Comment by bunnyvishal6** [comment]
@h2qutc getting this error in version 4.0.1

```
vendor.js:33913 ERROR TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.
    at subscribeTo (vendor.js:139880)
    at innerSubscribe (vendor.js:130889)
    at MergeMapSubscriber._innerSub (vendor.js:135230)
    at MergeMapSubscriber._tryNext (vendor.js:135224)
    at MergeMapSubscriber._next (vendor.js:135207)
    at MergeMapSubscriber.next (vendor.js:130442)
    at Observable._subscribe (vendor.js:139899)
    at Observable._trySubscribe (vendor.js:129898)
    at Observable.subscribe (vendor.js:129884)
    at MergeMapOperator.call (vendor.js:135192)
```

**Comment by rscherer** [comment]
@h2qutc Thanks for your effort to release this update. However, I'm still getting a **Class constructor NgxMatNativeDateAdapter cannot be invoked without 'new'** error. 

I'm getting this error using Angular Material 10.1.1. I tried to use the Material 9.2.4 + v4.0.1 and got some compilation errors like **"export 'DateRange' was not found in '@angular/material/datepicker'**, and using Material 10.1.x + 2.0.4 still results in the 'without new' error.

**Comment by bunnyvishal6** [comment]
@h2qutc Thanks a lot for this component, any update on this?

**Comment by h2qutc** [comment]
@bunnyvishal6  A new version will be released tomorrow to resolve this issue.

**Comment by bunnyvishal6** [comment]
Awesome, will be waiting.

**Comment by bunnyvishal6** [comment]
@h2qutc  any update on this? Why master is not updated with latest commits? If any documentation is there I would also love to try contributing to this repo.

**Comment by h2qutc** [comment]
@bunnyvishal6  Sorry for late reply. A new version v4.0.2 has already been released to resolve the problem "vendor.js:33913 ERROR TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable...."

For your question, there is a [Live demo and documentation](https://h2qutc.github.io/angular-material-components/datetimepicker) but it will be needed to update. I will try to do that and provide a Changelog.md.

**Comment by celian-garcia** [comment]
> @bunnyvishal6 Sorry for late reply. A new version [@angular-material-components/datetime-picker](https://www.npmjs.com/package/@angular-material-components/datetime-picker) has already been released to resolve the problem "vendor.js:33913 ERROR TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable...."
> 
> For your question, there is a [Live demo and documentation](https://h2qutc.github.io/angular-material-components/datetimepicker) but it will be needed to update. I will try to do that and provide a Changelog.md.

@h2qutc Strangely the direct link to datetime-picker demo is not working, but root link is redirecting anyway https://h2qutc.github.io/angular-material-components

**Comment by DantonGodoy** [comment]
> @llyrowenjones a new version @angular-material-components/datetime-picker@4.0.0 has been released to fix the above bugs
> 
> ```
> npm install --save @angular-material-components/datetime-picker@4.0.0
> ```
> 
> @gultyaev The peer dependencies is also fixed in the version 4.0.0.

Hello. But version 4+ has no support for Angular 9. What to do??
